### ISSUE TITLE
Format error decoding WebP: An expected chunk was missing

### ISSUE THREAD

**Issue by francisdb** [image-rs/image-webp#73]
Reproducer with attached file

```rust
let decoder = image::io::Reader::open(&file_path)?.with_guessed_format()?;
let (width_file, height_file) = decoder.into_dimensions();
```

gives me

```
Format error decoding WebP: An expected chunk was missing
```

The file I'm trying to read opens properly in the os preview / gimp.
[InstructionCard1.webp.zip](https://github.com/image-rs/image/files/15260977/InstructionCard1.webp.zip)

**Comment by fintelia** [comment]
Thanks for reporting this!

Do you happen to know where this image came from? Based on the WebP spec, I thought that all unknown chunks were required to go at the end of the file but this image contains a 78 byte PSAI chunk before the EXIF chunk. The decoder stops looking for chunks once it encounters an unknown chunk (and PSAI isn't one of the known chunks) and thus concludes that the EXIF chunk is missing. That causes the error you see. 

I haven't been able to find any info online about the PSAI chunk, but the contents contain a couple instances of 8BIM which is associated with Photoshop [Image Resource Blocks](https://www.adobe.com/devnet-apps/photoshop/fileformatashtml/#50577409_pgfId-1037685) and the structure seems to match up with that description. The chunk contains ResolutionInfo, pixel aspect ratio, and a third field (`04 43`) that isn't documented there

**Comment by francisdb** [comment]
Will try to contact the designer. Thanks for looking into this.

**Comment by francisdb** [comment]
The image was indeed saved with Photoshop

<img width="478" alt="image" src="https://github.com/image-rs/image-webp/assets/161305/dbeb451d-3700-4484-8d6e-75adad3e5ea2">

**Comment by fintelia** [comment]
Thanks for sharing that dialog. The most interesting part to me is that the image has EXIF and PSAI chunks even though all the checkboxes were _unchecked_. The EXIF chunk was at least empty, but the PSAI chunk actually had a small amount of content...

**Comment by francisdb** [comment]
Confirming that this is fixed in `0.1.3`
